agent/dns.go

```diff
@@ -195,7 +202,7 @@ func (d *DNSServer) handleQuery(resp dns.ResponseWriter, req *dns.Msg) {
 	// Setup the message response
 	m := new(dns.Msg)
 	m.SetReply(req)
-	m.Compress = !d.config.DisableCompression
+	m.Compress = !d.disableCompression.Load().(bool)
```

Comment:
I guarantee that in the `NewDNSServer` constructor. If you would forget this the server would crash immediately and you'd notice this pretty quick. Also, you have to do this only once for this to not fail so I think this approach is ok.